plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DataFileInfo.java

```diff
@@ -38,7 +38,9 @@ public enum DataFileType
 
     public DataFileInfo
     {
+        requireNonNull(path, "path is null");
         requireNonNull(dataFileType, "dataFileType is null");
+        requireNonNull(partitionValues, "partitionValues is null");
```

Comment:
We usually add `ImmutableList.copyOf` instead:
```suggestion
        partitionValues = ImmutableList.copyOf(partitionValues);
```

Reply:
There will be null value in the `partitionValues`

Reply:
Let's add a code comment, then.

Reply:
Noted, thanks